homeassistant/components/zha/core/channels/closures.py

```diff
@@ -35,6 +56,63 @@ def attribute_updated(self, attrid, value):
                 f"{self.unique_id}_{SIGNAL_ATTR_UPDATED}", attrid, attr_name, value
             )
 
+    async def async_set_user_code(self, code_slot: int, user_code: str) -> None:
+        """Set the user code for the code slot."""
+
+        set_pin_code = self.__getattr__("set_pin_code")
+        await set_pin_code(
```

Comment:
why not just `await self.set_pin_code()` ?

Reply:
is that a thing that will work? If so I can update to that easily

Reply:
definitely test it, but it should work.

Reply:
I'm still new to python :)
running tox now, will try with my lock in a bit

Reply:
Work got crazy for a couple weeks, but I finally was able to test with my door locks. As you suggested, it worked just fine. Updated commit pushed.
There is more that *could* be done... we can add more API tests and add UI to the device page, but this should be ready for the beta as an MVP.